src/ensembl/src/content/app/browser/track-panel/track-panel-storage-service.test.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  TrackPanelStorageService,
+  StorageKeys
+} from './track-panel-storage-service';
+import { ImageButtonStatus } from 'src/shared/image-button/ImageButton';
+import { Bookmark } from './trackPanelState';
+
+const mockStorageService = {
+  get: jest.fn(),
+  save: jest.fn(),
+  update: jest.fn(),
+  remove: jest.fn(),
+  clearAll: jest.fn()
+};
+
+const bookmarks = {
+  homo_sapiens_GCA_000001405_27: [
+    {
+      genome_id: 'homo_sapiens_GCA_000001405_27',
+      object_id: 'homo_sapiens_GCA_000001405_27:gene:ENSG00000139618',
+      object_type: 'gene',
+      label: 'BRCA2',
+      location: {
+        chromosome: '13',
+        end: 32400266,
+        start: 32315086
+      },
+      trackStates: {
+        'genes-transcripts': {
+          'gene-pc-fwd': ImageButtonStatus.INACTIVE
+        },
+        main: {
+          'gene-feat': ImageButtonStatus.INACTIVE,
+          'gene-feat-1': ImageButtonStatus.INACTIVE
+        }
+      }
+    }
+  ]
+};
```

Comment:
If all that these test do is check that a particular `mockStorageService` method gets called with a particular argument, then what does it matter whether the argument has a shape real bookmarks? Anything would do, just type it as `any`.

Reply:
That's done